differential-expression/rnaseq_DGE.Rmd

```diff
@@ -0,0 +1,216 @@
+`%>%` <- dplyr::`%>%`
+
+# Attach libraries
+library(limma)
+library(edgeR)
+```
+
+Create output folders.
+
+```{r}
+# Create the results folder if it doesn't exist
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+
+# Create the plots folder if it doesn't exist
+if (!dir.exists("plots")) {
+  dir.create("plots")
+}
+```
+
+### Read in refine.bio data
+
+```{r Import data from .tsv files}
+# Read in metadata tsv file
+metadata.df <- readr::read_tsv(file.path("data", "metadata_SRP078441.tsv")) %>%
+  # Get rid of columns that have only NAs
+  dplyr::select(-which(apply(is.na(.), 2, all)))
+
+# Read in data tsv file
+expression.df <- readr::read_tsv(file.path("data", "SRP078441.tsv")) %>%
+  tibble::column_to_rownames("Gene") # Store gene names as the rownames for now
+```
+
+The information we need to make the comparison is in the `refinebio_title` 
+column of the metadata data.frame.
+
+```{r}
+head(metadata.df$refinebio_title)
+```
+
+We'll use some tidyverse functionality to extract this information into it's
+own column to make it easier to use.
+
+```{r}
+metadata.df <- metadata.df %>%
+  # the last bit of the title, separated by "-" contains the mutation 
+  # information that we want to extract
+  dplyr::mutate(mutation_status = stringr::word(refinebio_title,
+                                                -1, sep = "-")) %>%
+  # now let's make this simpler
+  dplyr::mutate(mutation_status = dplyr::case_when(
+    grepl("ASXL1|ASXL2", mutation_status) ~ "mutation",
+    grepl("ASXLwt", mutation_status) ~ "wildtype"
+  )) 
+```
+
+Let's take a look at `metadata.df` to see if this worked.
+
+```{r}
+# looking at the first 6 rows of the metadata_df and only at the columns that
+# contain the title and the mutation status we extracted from the title
+head(dplyr::select(metadata.df, refinebio_title, mutation_status))
+```
+
+## Differential expression analysis
+
+Now that we've prepped our data, we're ready to test for differentially 
+expressed genes.
+
+Let's make sure samples are in the same order.
+
+```{r}
+all.equal(colnames(expression.df), metadata.df$refinebio_accession_code)
+```
+
+```{r}
+# put samples in ascending orderaccording to their accession code in the 
+# metadata
+metadata.df <- metadata.df %>%
+  dplyr::arrange(refinebio_accession_code)
+
+# use the metadata.df information to reorder the columns of the expression
+# data frame
+expression.df <- expression.df[, metadata.df$refinebio_accession_code]
+```
+
+Check again.
+
+```{r}
+all.equal(colnames(expression.df), metadata.df$refinebio_accession_code)
+```
+
+### limma-voom
+
+Here, we're following the steps in [this section](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#limma-voom) 
+of the [tximport vignette](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html).
+
+```{r}
+# construct a DGEList object from the refine.bio output
+y <- DGEList(as.matrix(expression.df))
```

Comment:
```suggestion
y <- DGEList(as.matrix(expression.df))
```
I get that you are just copying from the tximport vignette, but why did they name it `y`? That doesn't tell me anything about the object. Can we call it something that explains more about what it is?

Reply:
Or if we can explain why we are naming this object `y`.